pkg/operation/botanist/etcdencryption.go

```diff
@@ -0,0 +1,332 @@
+package botanist
+
+import (
+	"fmt"
+
+	gardenv1beta1 "github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+	"github.com/gardener/gardener/pkg/operation/common"
+	encryptionconfiguration "github.com/gardener/gardener/pkg/operation/etcdencryption"
+	"github.com/gardener/gardener/pkg/utils"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiserverconfigv1 "k8s.io/apiserver/pkg/apis/config/v1"
+	"k8s.io/client-go/util/retry"
+)
+
+// CreateEtcdEncryptionConfiguration creates a secret
+func (b *Botanist) CreateEtcdEncryptionConfiguration() error {
+	b.Logger.Info("starting CreateEtcdEncryptionConfiguration")
+
+	enableEtcdEncryption, err := utils.CheckVersionMeetsConstraint(b.Shoot.Info.Spec.Kubernetes.Version, ">= 1.13")
+	if err != nil {
+		return err
+	}
+	if !enableEtcdEncryption {
+		b.Logger.Info("support for etcd encryption is offered automatically starting with kubernetes version 1.13")
+		return nil
+	}
+	needToWriteConfig := false
+	exists, ecSeed, forcePlaintext, err := b.readEncryptionConfigurationFromSeed()
+	if err != nil {
+		return err
+	}
+	if !exists {
+		// create new passive EncryptionConfiguration if it does not exist yet
+		// and remember to write this created configuration
+		b.Logger.Info("no EncryptionConfiguration found in seed. Creating new passive configuration.")
+		ecSeed, err = encryptionconfiguration.CreateNewPassiveConfiguration()
+		if err != nil {
+			return err
+		}
+		needToWriteConfig = true
+	} else {
+		exists, ecGarden, err := b.readEncryptionConfigurationFromGarden()
+		if err != nil {
+			return err
+		} else if !exists {
+			return fmt.Errorf("EncryptionConfiguration exists in seed but not in garden")
+		}
+		// if it exists already, it needs to be consistent
+		consistent, err := b.isEncryptionConfigurationConsistent(ecSeed, ecGarden)
+		if (err != nil) || !consistent {
+			return err
+		}
+		if forcePlaintext {
+			// if annotation requires plaintext secrets, thou shalt get plaintext secrets
```

Comment:
typo?

Reply:
is probably more flowery old-style speech ... 
https://www.linguee.de/englisch-deutsch/uebersetzung/thou+shalt.html ... I can remove it if you want ...

Reply:
adapted comment